op-e2e/actions/interop/emitter_contract_test.go

```diff
@@ -0,0 +1,216 @@
+package interop
+
+import (
+	"crypto/ecdsa"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/devkeys"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/interop/contracts/bindings/emit"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/interop/contracts/bindings/inbox"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/event"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	stypes "github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+type userWithKeys struct {
+	key     devkeys.ChainUserKey
+	secret  *ecdsa.PrivateKey
+	address common.Address
+}
+
+func TestEmitterContract(gt *testing.T) {
+	var (
+		t      = helpers.NewDefaultTesting(gt)
+		is     *InteropSetup
+		actors *InteropActors
+		aliceA *userWithKeys
+		aliceB *userWithKeys
+		emitTx *types.Transaction
+	)
+	resetTest := func() {
+		is = SetupInterop(t)
+		actors = is.CreateActors()
+		aliceA = setupUser(t, is, actors.ChainA, 0)
+		aliceB = setupUser(t, is, actors.ChainB, 0)
+		initializeChainState(t, actors)
+		emitTx = initializeEmitterContractTest(t, aliceA, actors)
+
+		// TODO: This is required otherwise the exec msg will fail with
+		// a timestamp invariant issue
```

Comment:
```suggestion
		// This is required otherwise the exec msg will fail with
		// a timestamp invariant issue
```

Are we keeping it like this, cancel the TODO?